### ISSUE TITLE
Wrong struct format of SceNetAdhocctlParams

### ISSUE THREAD

**Issue by ANR2ME** [pspdev/pspsdk#46]
SceNetAdhocctlParams struct have a wrong format, it should have the `bssid` and `nickname` position swapped (at least on newer firmware, not sure about old firmware) otherwise games like _Ys vs. Sora no Kiseki_ will shows the wrong game room name (gibberish/random letters) taken from mac address (bssid) instead of the player's nickname when a PSP emulator is using the wrong struct format like this:
```
struct SceNetAdhocctlParams
{
	/** Channel number */
	int channel;
	/** Name of the connection */
	char name[8];
	/** The BSSID */
	unsigned char bssid[6];
	/** Nickname */
	char nickname[128];
};
```
PS: This was tested by comparing the struct contents taken from _Ys vs. Sora no Kiseki_ on JPCSP using prx files from official firmware 6.xx

Should have similar format to what VitaSdk have:
```
typedef struct SceNetAdhocctlParameter {
     int channel;
     struct SceNetAdhocctlGroupName groupName;
     struct SceNetAdhocctlNickname nickname;
     struct SceNetAdhocctlBSSId bssid;
 } SceNetAdhocctlParameter;
```

**Comment by fjtrujy** [comment]
Hello @ANR2ME,
This is a really interesting question......
I have some "documentation around with version 6.60" and they show in the same format that we have, having `nickname` as the latest option.

However, if we take a look at `PPSSPP` they have :
https://github.com/hrydgard/ppsspp/blob/9271db09163d7ae3becd35c73decab0b3acad6d3/Core/HLE/proAdhoc.h#L284-L290
```c
// Active Virtual Network Information (Adhoc Group Host/Creator's device info, similar to AP?)
typedef struct SceNetAdhocctlParameter {
  s32_le channel;
  SceNetAdhocctlGroupName group_name; // This group name is probably similar to SSID name on AP
  SceNetAdhocctlNickname nickname; // According to the old PSPSDK this is the bssid, but according to the dumped content when using newer firmware this is the nickname (this is also the nickname on VitaSDK)
  SceNetAdhocctlBSSId bssid; // FIXME: bssid and nickname position might be swapped on older/newer firmware?
} PACK SceNetAdhocc	tlParameter;
```
As you can see there is a funny comment `// FIXME: bssid and nickname position might be swapped on older/newer firmware?`

So actually I'm not sure about what values we should put

**Comment by ANR2ME** [comment]
Hi @fjtrujy thanks for replying.

I'm the one that made that `// FIXME:` comment, because there is a game (Ys vs. Trails in the Sky Alternative Saga / NPJH50276) that shows a strange game room name as MAC address instead of Player's Nickname, which is wrong, but after swapping those fields it shows the game room correctly https://github.com/hrydgard/ppsspp/issues/13493#issuecomment-701710246

That game is for 6.20 firmware (based on it's PARAM.SFO), and i'm not sure whether the field was swapped on older firmware or not, which is why i added that `FIXME` comment for someone to confirm it.
So far there are no other games affected by those swapped fields, so i guess it's the correct order.

**Comment by diamant3** [comment]
I want to try this issue, but I have only one PSP. Can I use this link: https://forums.ppsspp.org/showthread.php?tid=8349 to test it? The next version of firmware is 6.20, am I right? (By the way, I'm running 6.60 pro-c, but I will downgrade it to 6.20.) Do I need to use both the new and old are official firmware? thanks.

**Comment by ANR2ME** [comment]
No, it should be older than 6.20 to be compared with newest fw, since the game that known to use the new struct was designed for 6.20 fw

**Comment by diamant3** [comment]
Thanks for your response! I will try to do some research about that, but I don't think I can take this. Maybe other people can solve this. Thanks again!

**Comment by sharkwouter** [comment]
This has been resolved by #126
